dialdbg/src/main.rs

```diff
@@ -0,0 +1,239 @@
+        .allow_downgrade()
+        .connect()
+        .await
+    {
+        Ok(c) => match c {
+            ViamChannel::WebRTC(c) => Some(c.get_stats().await),
+            _ => None,
+        },
+        Err(e) => {
+            log::error!("{}: {e}", strings::DIAL_ERROR_PREFIX);
+            None
+        }
+    }
+}
+
+fn output_connection_stats(stats: stats::StatsReport, out: &mut Box<dyn io::Write>) -> Result<()> {
+    // StatsReport contains 13 types of stat reports; there may be more relevant stats to print
+    // here, but for now I have stuck with only printing the candidates.
+    writeln!(out, "\nnominated ICE candidates:\n")?;
+    let now = Instant::now();
+    for value in stats.reports.into_values() {
+        match value {
+            stats::StatsReportType::LocalCandidate(lcs) => {
+                writeln!(out, "\tlocal ICE candidate:")?;
+                writeln!(out, "\t\tIP address: {}", lcs.ip)?;
+                writeln!(out, "\t\tport: {}", lcs.port)?;
+                writeln!(
+                    out,
+                    "\t\tnominated {:#?} ago",
+                    now.duration_since(lcs.timestamp)
+                )?;
+                writeln!(out, "\t\trelay protocol: {}", lcs.relay_protocol)?;
+                writeln!(out, "\t\tnetwork type: {}", lcs.network_type)?;
+            }
+            stats::StatsReportType::RemoteCandidate(rcs) => {
+                writeln!(out, "\tremote ICE candidate:")?;
+                writeln!(out, "\t\tIP address: {}", rcs.ip)?;
+                writeln!(out, "\t\tport: {}", rcs.port)?;
+                writeln!(
+                    out,
+                    "\t\tnominated {:#?} ago",
+                    now.duration_since(rcs.timestamp)
+                )?;
+                writeln!(out, "\t\trelay protocol: {}", rcs.relay_protocol)?;
+                writeln!(out, "\t\tnetwork type: {}", rcs.network_type)?;
+            }
+            _ => {}
+        }
+    }
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args = Args::parse();
+
+    let mut uri = args.uri.unwrap_or_default();
+    let mut secret = args.secret.unwrap_or_default();
+    if args.config.is_some() {
+        (uri, secret) = read_config(args.config.unwrap())?;
+    }
+
+    // Write to output file or STDOUT if none is provided.
+    let mut out: Box<dyn io::Write> = match args.output {
+        Some(output) => match fs::File::create(output) {
+            Ok(output_file_writer) => Box::new(output_file_writer),
+            Err(e) => {
+                return Err(anyhow!("error opening --output file: {e}"));
+            }
+        },
+        None => Box::new(io::stdout()),
+    };
+
+    let mut log_config_setter = None;
+    if !args.nogrpc {
+        writeln!(out, "\nDebugging dial with basic gRPC...\n")?;
+        // Start logger with Debug-level logging and append logs to a file in a temp directory.
+        let log_path = std::env::temp_dir().join("grpc_temp.log");
```

Comment:
Given your bug you showed me earlier about these files not being deleted, putting something like
```
if log_path.try_exists() == Ok(true) {
   // remove log_path
}
```
might not be a bad idea in case a previous invocation errors out before this file is deleted.
https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.try_exists